python/cuda_parallel/tests/test_reduce.py

```diff
@@ -550,3 +551,30 @@ def binary_op(x, y):
     d_in = cp.zeros(size)[::2]
     with pytest.raises(ValueError, match="Non-contiguous arrays are not supported."):
         _ = algorithms.reduce_into(d_in, d_out, binary_op, h_init)
+
+
+def test_reduce_struct_type():
```

Comment:
Q: why do we repeat the same test here? I thought `test_reduce_api.py` also gets tested?

Reply:
Yes, but at the moment the way our [documentation examples](https://github.com/NVIDIA/cccl/blob/2c5bdcd8691c2917c06c7d4c570a3082e0180660/python/cuda_parallel/cuda/parallel/experimental/algorithms/reduce.py#L162) work is by `literalinclude`'ing the tests from `test_reduce_api.py`. 

- There's no _technical_ reason we can't just literalinclude the tests from `test_reduce.py` directly. But I do like keeping all the examples together in a separate file. 
- I'd like to move away from this to proper doctests (although maybe that should wait until our documentation system change away from repo_man). cc: @NaderAlAwar for viz as we briefly chatted about that as well.

Reply:
Same question asked differently: What is lost if we delete this copy? (Why do you want it here, too?)

Reply:
- tests in this module are primarily for testing correctness
- tests in the module `test_reduce_api.py` are didactic and primarily for _teaching_ the API

We might choose to change the tests in `test_reduce_api.py` so that they illustrate a concept more clearly, or they might disappear altogether in favor of doctests. Tests in this module are far less likely to be deleted.

We could get rid of `test_reduce_api.py` altogether and instead have the documentation reference tests directly from this module, but again, it seems cleaner to have all the documentation examples together in a single module.

Reply:
FWIW, I had terrible run-ins with non-trivial copy-pasted code. E.g. I thought I fixed the problem already not realizing there are two copies, or being confused about which is which in the heat of the moment, etc.

My strong preference would be to delete the copy here. As long as you have me as a reviewer, there will be someone asking a lot of questions if you want to remove it completely without giving a good explanation :-)

I.e. if/when you get rid of test_reduce_api.py, I'd be paranoid about making sure no tests are losts.